velox/exec/Window.h

```diff
@@ -60,13 +60,21 @@ class Window : public Operator {
   }
 
  private:
+  // Used for k preceding/following frames. Index is the column index if k is a
+  // column. value is used to read the column values from the input, or contains
+  // the constant value if k is a constant.
+  struct FrameChannelArg {
+    column_index_t index;
+    VectorPtr value;
+  };
+
   // Structure for the window frame for each function.
   struct WindowFrame {
     const core::WindowNode::WindowType type;
     const core::WindowNode::BoundType startType;
     const core::WindowNode::BoundType endType;
-    const std::optional<column_index_t> startChannel;
-    const std::optional<column_index_t> endChannel;
+    const std::optional<FrameChannelArg> startFrameArg;
```

Comment:
Please add a comment here that these are only set for k Preceding and k Following cases.